### PR TITLE
Added a Function for DocArrayInMemorySearch

### LINKED ISSUE #2: RAG Implementation: Document Retrieval 🚀
### **Project Overview** 
For this project we are looking at creating a RAG solution leveraging the power of Microsoft Azure that takes in documents from users and allow user to query the documents through a voice message. 

### **What is Needed**
We need contributors to implement a proper and efficient chunking strategy for the documents uploaded by user so as to ensure that the most relevant section of the document is retrieved to answer queries.
- Consider breaking the documents into smaller, semantically meaningful chunks for optimal retrieval.
- The solution should scale efficiently and work with the suggested document types (PDFs, TXT files).

### **How to Contribute**
1. Get familiar with the starter scripts and documentations already provided in the [repository](https://github.com/mlsanigeria/speak-to-docs).
2. **Test your code:** It is pertinent that you check your code runs smoothly before making a pull request. 
3. **Create a Pull Request:** Submit a pull request to the repository with your code in the appropriate script and include any necessary documentation

### **Getting Started**
Ensure to properly read through the [Contribution Guidelines](https://github.com/mlsanigeria/speak-to-docs/blob/9240693ec9a595a1b52fbd69aa187ea545eae65d/CONTRIBUTING.md) to get started.

We are looking forward to your contributions. Happy Hacking! 🎉

### LINKED ISSUE #4: RAG Implementation: Efficient Memory Storage 🧠
### **What is Needed**
Contributors are needed to implement an efficient memory storage that will help to optimally tracks conversation history, even in scenarios involving multiple file uploads. Maintaining proper session state management. Session timeouts are an option. 

### **How to Contribute**
1. Get familiar with the scripts and documentations already provided in the [repository](https://github.com/mlsanigeria/speak-to-docs). 
2. **Test your code:** It is pertinent that you check your code runs smoothly before making a pull request.
3. **Create a Pull Request:** Submit a pull request to the repository with your code in the appropriate script and include any necessary documentation. 

### **Getting Started**
Before you begin, ensure you have read the [Contribution Guidelines](https://github.com/mlsanigeria/speak-to-docs/blob/9240693ec9a595a1b52fbd69aa187ea545eae65d/CONTRIBUTING.md) in the repository

We are excited to see your contributions! Happy Hacking! 🎉

### LINKED ISSUE #29: LangChain DocArrayInMemorySearch as Vector Store Implementation 📄
**What is Needed**
Contributors are needed to implement LangChain `DocArrayInMemorySearch` as a vector store. Fine-tune the [streamlit script](https://github.com/mlsanigeria/speak-to-docs/blob/main/main.py) to reflect this feature. Also ensure that document memory is cached per session, enabling the app to retain uploaded documents and queries during individual user sessions.

**How to Contribute**
1. Get familiar with the scripts and documentation already provided in the [repository](https://github.com/mlsanigeria/speak-to-docs).
2. Test your code: It is pertinent that you check your code runs smoothly before making a pull request.
3. Create a Pull Request: Submit a pull request to the repository with your code in the appropriate script and include any necessary 
documentation.

**Getting Started**
Before you begin, ensure you have read the [Contribution Guidelines](https://github.com/mlsanigeria/speak-to-docs/blob/9240693ec9a595a1b52fbd69aa187ea545eae65d/CONTRIBUTING.md) in the repository

We are excited to see your contributions! Happy Hacking! 🎉

### PR DESCRIPTION
**Pull request by Odeyiany2**
Hi @Sammybams, 
I have implemented a function that processes the results from the extracted contents using Azure Document Intelligence. The function performs the following tasks:

1. Chunking: Uses the previously implemented chunking function to split the extracted content into manageable pieces.
2. Embedding: Leverages the OpenAI embedding model to create embeddings for the chunked content.
3. Vector Store: Stores the resulting embeddings in a DocArrayInMemorySearch vector store.

Additionally, I have included a resource that helped guide my implementation. Could you please review the code and provide feedback? I would appreciate any suggestions or improvements.
Thank you!

### CODE REVIEW

File: src/rag_functions.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 from azure.core.credentials import AzureKeyCredential
 from langchain.text_splitter import RecursiveCharacterTextSplitter
 
+
```
Comment (Sammybams) [inline src/rag_functions.py]:
```suggestion
```

**Comment by Odeyiany2** [comment]
Hi @Sammybams This is how it works. I had to change the initialization of the open ai embedding model from `OpenAIEmbeddings` to `AzureOpenAIEmbeddings` the documentation has been updated so to mitigate the error of `resource not found` when calling the API key, the update had to be done. 
Screenshot 2024-10-21 142149

Screenshot 2024-10-21 141232

**Comment by Sammybams** [comment]
> Hi @Sammybams This is how it works. I had to change the initialization of the open ai embedding model from `OpenAIEmbeddings` to `AzureOpenAIEmbeddings` the documentation has been updated so to mitigate the error of `resource not found` when calling the API key, the update had to be done. Screenshot 2024-10-21 142149
> 
> Screenshot 2024-10-21 141232

Hi @Odeyiany2,
Kindly confirm what version of openai you are using. You shouldn't have that issue.

**Comment by Odeyiany2** [comment]
Screenshot 2024-10-21 170339
Screenshot 2024-10-21 170353
Hi @Sammybams I used the versions provided in the requirements.txt file and it worked smoothly. The versions I used previously were higher than what was expected which made to make the previous changes. I have made the changes and it still worked. Thank you.

**Review by Sammybams — APPROVED** [review]
Great work @Odeyiany2. Nice implementation.

**Pull request merged**